supabase/UserContext.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState, createContext, useContext } from 'react'
```

Comment:
Not sure if we can use this context provider in our Expo / React native implementation

Reply:
I think you can use it in React Native, but I haven't before.